programs/mp-sol-restaking/src/instructions/admin/configure_main_vault.rs

```diff
@@ -1,52 +1,104 @@
 use crate::{error::ErrorCode, state::MainVaultState, MAX_PERFORMANCE_FEE_BP, MAX_WITHDRAW_FEE_BP};
 use anchor_lang::prelude::*;
+use anchor_spl::token::{Mint, TokenAccount};
 
-#[derive(AnchorSerialize, AnchorDeserialize, Debug)]
-pub struct ConfigureMainVaultValues {
-    unstake_ticket_waiting_hours: Option<u16>,
-    withdraw_fee_bp: Option<u16>,
-    performance_fee_bp: Option<u16>,
-    treasury_mpsol_account: Option<Pubkey>,
-    new_admin_pubkey: Option<Pubkey>,
+#[derive(Accounts)]
+pub struct ConfigureTreasuryAccount<'info> {
+    #[account()]
+    pub admin: Signer<'info>,
+
+    #[account(mut, has_one = admin, has_one = mpsol_mint)]
+    pub main_state: Account<'info, MainVaultState>,
+
+    #[account()]
+    pub mpsol_mint: Box<Account<'info, Mint>>,
+
+    #[account(token::mint = mpsol_mint)]
+    pub treasury_mpsol_account: Account<'info, TokenAccount>,
+}
+pub fn handle_configure_treasury_account(ctx: Context<ConfigureTreasuryAccount>) -> Result<()> {
+    ctx.accounts.main_state.treasury_mpsol_account =
+        Some(ctx.accounts.treasury_mpsol_account.key());
+    Ok(())
+}
+
+#[derive(Accounts)]
+pub struct ConfigureUnstakeWaitingHours<'info> {
```

Comment:
Usually, I'm making single accounts struct reused by multiple ixes

Reply:
ok, doing that